os/fs/driver/mtd/smart.c

```diff
@@ -5446,24 +5446,24 @@ static int smart_journal_recovery(FAR struct smart_struct_s *dev, journal_log_t
 			}
 			
 			/* Data sector is released, checkout journal data */
-			return smart_journal_checkout(dev, log, address);
+			ret = smart_journal_checkout(dev, log, address);
```

Comment:
I have one question.
If crc fails, do we have to delete the sector?

Reply:
crc fails mean that written data is invalid so we have to drop it.